### PR TITLE
[Resource] Fix resource loading / metadata

### PR DESCRIPTION
**Pull request by isometriks**
| Q               | A
| --------------- | ---
| Bug fix?        | no|yes
| New feature?    | no|yes
| BC breaks?      | no|yes
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | MIT

### CODE REVIEW

**Comment by isometriks** [comment]
Taken from https://github.com/Sylius/Sylius/pull/5233#issuecomment-227219133

@pjedrzejewski Let me know if I need to do anything else for this. Seems to be all working.

File: src/Sylius/Bundle/ResourceBundle/EventListener/ODMMappedSuperClassSubscriber.php [hunk 1]
```diff
@@ -76,17 +76,25 @@ private function convertToDocumentIfNeeded(ClassMetadataInfo $metadata)
...
-
-            if (false === $this->isSyliusClass($parentMetadata)) {
+            if (false === in_array($parent, $configuration->getMetadataDriverImpl()->getAllClassNames())) {
                 continue;
             }
 
-            if (false === in_array($parent, $configuration->getMetadataDriverImpl()->getAllClassNames())) {
+            $parentMetadata = new ClassMetadata(
+                $parent,
+                $configuration->getNamingStrategy()
```
Comment (isometriks) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ODMMappedSuperClassSubscriber.php]: This was my only question. Do the tests check ODM as well? This line wasn't here before in ODM only in ORM

**Comment by isometriks** [comment]
ping @sylius/core @pjedrzejewski

**Comment by pjedrzejewski** [comment]
Thank you Craig, nice work! 👍

**Pull request merged**